src/peft/utils/save_and_load.py

```diff
@@ -305,6 +305,25 @@ def _find_mismatched_keys(
     return peft_model_state_dict, mismatched
 
 
+def _insert_adapter_name_into_state_dict(
```

Comment:
This is the same code as before, but factored out into a function so that it can be reused for hotswapping.